src/components/visx/TimeSeriesChart.tsx

```diff
@@ -141,17 +141,32 @@ export const TimeSeriesChart = <Datum extends {}>({
   const latestDatum = data?.[data.length - 1];
 
   // Chart state
+  const getClampedZoomDomain = useCallback(
+    (unclamped: number) => {
+      return clamp(
+        Math.max(1e-320, Math.min(Number.MAX_SAFE_INTEGER, unclamped)),
+        minZoomDomain,
+        xAccessor(latestDatum) - xAccessor(earliestDatum)
+      );
+    },
+    [earliestDatum, latestDatum, minZoomDomain, xAccessor]
+  );
+
   const [zoomDomain, setZoomDomain] = useState<number | undefined>(
-    defaultZoomDomain ?? xAccessor(latestDatum) - xAccessor(earliestDatum)
+    defaultZoomDomain
+      ? getClampedZoomDomain(defaultZoomDomain)
+      : xAccessor(latestDatum) - xAccessor(earliestDatum)
   );
 
   const [zoomDomainAnimateTo, setZoomDomainAnimateTo] = useState<number | undefined>();
 
   useEffect(() => {
     if (defaultZoomDomain) {
-      setZoomDomainAnimateTo(defaultZoomDomain);
+      const clampedZoomDomain = getClampedZoomDomain(defaultZoomDomain);
+      setZoomDomain(clampedZoomDomain);
+      setZoomDomainAnimateTo(clampedZoomDomain);
     }
-  }, [defaultZoomDomain]);
+  }, [defaultZoomDomain, getClampedZoomDomain]);
```

Comment:
do we want this to re-run whenever the getClampedZoomDomain dependencies change?

Reply:
hmm I guess the biggest implication here would be that if we get new data (`latestDatum` dep of `getClampedZoomDomain` updates), having this dep means that we re-zoom to `defaultZoomDomain` if it's set (user is not currently zooming). I think that makes sense to me, but is there something else I'm missing?

Reply:
it looks like this caused the PNL chart on portfolio to behave strangely when zooming after selecting a period, and removing `getClampedZoomDomain` fixes it :o

Reply:
oh it's cuz we were passing in an inline func to [xAccessor](https://github.com/dydxprotocol/v4-web/blob/main/src/views/charts/PnlChart.tsx#L213) in PnlChart which was being counted as "new" each render. Updating `PnlChart` to not always be passing in these new funcs/objs